src/VisualStudio/Core/Impl/SolutionExplorer/SourceGeneratedFileItems/SourceGeneratedFileItemSource.cs

```diff
@@ -154,11 +158,7 @@ public void BeforeExpand()
         {
             lock (_gate)
             {
-                // We should not have an existing computation active
-                Contract.ThrowIfTrue(_cancellationTokenSource is not null);
```

Comment:
Can we add something like CancellationSeries.AssertIsCancelled() as a way to bring this assert back in some form? This assert is potentially going to catch cases where we collapsed but failed to cancel off the prior work, which would indicate we potentially kept refreshing this in the background in perpetuity.

Reply:
Alternatively, just make CancellationSeries be nullable with the existing pattern which will make it a bit easier. It communicates that when expanded there's a series of cancellations, but once closed there shouldn't be a request for a new token.

Reply:
> Alternatively, just make CancellationSeries be nullable with the existing pattern which will make it a bit easier

CancellationSeries exists almost for the sole purpose of eliminating `null` as  potential state, so I definitely want to avoid going back to that. I'll consider the other items soon.

Reply:
➡️ Restored this assertion through `CancellationSeries`